src/server/scripts/Northrend/zone_borean_tundra.cpp

```diff
@@ -710,16 +710,26 @@ class npc_captured_beryl_sorcerer : public CreatureScript
                 switch (eventId)
                 {
                     case EVENT_ADD_ARCANE_CHAINS:
-                        if (Player* summoner = me->ToTempSummon()->GetSummonerUnit()->ToPlayer())
+                        if (TempSummon* tempSummon = me->ToTempSummon())
                         {
-                            summoner->CastSpell(summoner, SPELL_ARCANE_CHAINS_CHANNEL_II, TriggerCastFlags(TRIGGERED_FULL_MASK & ~TRIGGERED_IGNORE_AURA_INTERRUPT_FLAGS & ~TRIGGERED_IGNORE_CAST_ITEM & ~TRIGGERED_IGNORE_POWER_AND_REAGENT_COST & ~TRIGGERED_IGNORE_GCD));
-                            _events.ScheduleEvent(EVENT_FOLLOW_PLAYER, 1s);
+                            if (Unit* summonerUnit = tempSummon->GetSummonerUnit())
+                            {
+                                if (Player* summoner = summonerUnit->ToPlayer())
```

Comment:
Conversion to player here might be entirely unnecessary

Reply:
Thanks